docs/develop/topics/theming.md

```diff
@@ -0,0 +1,58 @@
+# Theming 
+
+This page will provide you with in-depth information about the Invenio RDM theme structure, which is inheriting the concepts from the Semantic UI
+
+## Semantic UI
+
+### Concepts
+
+From semantic UI docs:
+
+> Semantic UI treats words and classes as exchangeable concepts.
+Classes use syntax from natural languages like noun/modifier relationships, word order, and plurality to link concepts intuitively.
+
+It basically refers to using the CSS classes to describe how the element looks like, not what it is, f.e.
+
+```html
+<div class="ui label right floated"></div>
+```
+
+tells us more about what to expect from the final view, rather than
+
+```html
+<div class="metadata-label"></div>
+```
+
+which has significant influence on the maintainability of the styling - and it is very important in big projects. It is easier for any developer to remember a few "behavioral" classes, rather than to remember that metadata-label will be always right floated.
+
+### Variety of available components and elements
+
+Semantic UI exists both in "clean" HTML/CSS as well as in React. To know more about best practices between the two, visit [Best practices](../../develop/best-practices/react.md) page.
+
+Semantic UI provides the developer with a plethora of ready to use React components and CSS classes. The important step in mastering the Semantic UI development is consulting the documentation and experimenting with the outcome of using the provided elements. Custom CSS rules should be the last resort of achieving the specific styling. 
+
+
+### Theme inheritance
+
+The theming is based on the inheritance system created by semantic UI, which needed to be adapted for Invenio RDM. The default structure is designed as follows:
+
+![Default theming inheritance](../img/theming-default-structure.png)
+
+While, for RDM we tweaked the theming structure into:
+
+![Default theming inheritance](../img/theming-rdm-structure.png)
+
+The change was achieved by defining our internal Invenio RDM theme inheritance rules, available in [theme.less](https://github.com/inveniosoftware/invenio-app-rdm/blob/6249106ba962514338f8e313ff033f8f1bbd3fce/invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/theme.less)
+
+#### Your own theme
```

Comment:
I am not sure I understand very well this last part, reading it. We can always improve it later.